src/Symfony/Component/HttpKernel/Controller/ArgumentResolver/RequestValueResolver.php

```diff
@@ -24,6 +25,17 @@ final class RequestValueResolver implements ValueResolverInterface
 {
     public function resolve(Request $request, ArgumentMetadata $argument): array
     {
-        return Request::class === $argument->getType() || is_subclass_of($argument->getType(), Request::class) ? [$request] : [];
+        $result = Request::class === $argument->getType() || is_subclass_of($argument->getType(), Request::class) ? [$request] : [];
+        if (!empty($result)) {
+            return $result;
+        }
+
+        $className = explode('\\', $argument->getType() ?? '');
+        $className = end($className);
+        if ('Request' === $className) {
```

Comment:
You could use `str_ends_with($className, '\\Request'` instead of this exploding and then extracting the last part.